### ISSUE TITLE
False positive with for boolean expressions with ?? operator

### ISSUE THREAD

**Issue by leiboldm** [dart-lang/linter#357]
```
class Foo {
  bool bar;
  void sayHello() {
    if (bar ?? false) print('hello');
  }
}
```

For this code, the condition in sayHello erroneously shows linter message "Conditions should not unconditionally evaluate to true or false."
